extensions.c

```diff
@@ -263,14 +263,18 @@ hibaext_maybe_deflate(struct sshbuf *in, int compress, struct sshbuf *out) {
 			}
 			sshbuf_put(temp, buf, stream.total_out);
 		}
+		deflateEnd(&stream);
 		ret = HIBA_OK;
 		debug3("hibaext_maybe_deflate: compression out %zu.", sshbuf_len(temp));
 	}
 err:
 #endif  /* WITH_EXTENSION_COMPRESSION */
 
 	if (ret == HIBA_OK) {
-		return sshbuf_putb(out, temp);
+		ret = sshbuf_putb(out, temp);
+		if (temp != in)
```

Comment:
Shouldn't it also be free'd for ret != HIBA_OK?

Reply:
Done